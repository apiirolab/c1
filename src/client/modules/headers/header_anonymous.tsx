import * as React from 'react';
import logo = require('./logo.png');

import { Menu, MenuProps } from 'semantic-ui-react';
import styled, { StyledComponentClass } from 'styled-components';

import { Logo } from './header_logo';

const TopPanel = styled.div`
  background: #efefef;
`;

const TopMenu: StyledComponentClass<MenuProps, {}> = styled(Menu)`
  &&&&& {
    position: fixed;
    width: 100%;
    border: 0px;
    border-radius: 0px;
    background: inherit;
    box-shadow: inherit;
    margin-bottom: 12px;
  }
`;

export const HeaderAnonymous = () => (
  <TopPanel>
    <TopMenu borderless>
      <Menu.Item>
        <Logo src={logo} alt="logo" />
      </Menu.Item>
    </TopMenu>
  </TopPanel>
);
