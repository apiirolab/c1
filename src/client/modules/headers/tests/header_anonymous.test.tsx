import * as React from 'react';

import { Provider } from 'mobx-react';
import { create } from 'react-test-renderer';

import { HeaderAnonymous } from '../header_anonymous';

describe('Headers', () => {
  describe('Anonymous', () => {
    const Header = (
      <Provider store={{}}>
        <HeaderAnonymous />
      </Provider>
    );

    it('renders correctly', () => {
      const root = create(Header);
      expect(root).toMatchSnapshot();
    });

    return {
      component: Header
    };
  });
});
