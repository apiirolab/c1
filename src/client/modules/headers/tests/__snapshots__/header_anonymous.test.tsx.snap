// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers Anonymous renders correctly 1`] = `
<div
  className="sc-bwzfXH kRBEAi"
>
  <div
    className="ui borderless sc-htpNat fHSfNu menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="sc-bdVaJa kPmfWy"
        src="file"
      />
    </div>
  </div>
</div>
`;
